Mixpanel/MixpanelExceptionHandler.m

```diff
@@ -33,38 +48,86 @@ - (instancetype)init {
     if (self) {
         // Create a hash table of weak pointers to mixpanel instances
         _mixpanelInstances = [NSHashTable weakObjectsHashTable];
-        
         // Save the existing exception handler
         _defaultExceptionHandler = NSGetUncaughtExceptionHandler();
         // Install our handler
         NSSetUncaughtExceptionHandler(&mp_handleUncaughtException);
+        // Install signal Handler
+        registerFatalSignals();
     }
     return self;
 }
 
 - (void)addMixpanelInstance:(Mixpanel *)instance {
     NSParameterAssert(instance != nil);
-    
     [self.mixpanelInstances addObject:instance];
 }
 
 static void mp_handleUncaughtException(NSException *exception) {
     MixpanelExceptionHandler *handler = [MixpanelExceptionHandler sharedHandler];
-    
-    // Archive the values for each Mixpanel instance
-    for (Mixpanel *instance in handler.mixpanelInstances) {
-        // Since we're storing the instances in a weak table, we need to ensure the pointer hasn't become nil
-        if (instance) {
-            [instance archive];
+    NSLog(@"!!!!!!");
+    if(isAlreadyExceptionOccured==0) {
+        OSAtomicOr32Barrier(1, &isAlreadyExceptionOccured);
+        // Archive the values for each Mixpanel instance
+        for (Mixpanel *instance in handler.mixpanelInstances) {
+            // Since we're storing the instances in a weak table, we need to ensure the pointer hasn't become nil
+            if (instance) {
+                [instance archive];
+            }
         }
+        
+        MixpanelError(@"Encountered an uncaught exception. All Mixpanel instances were archived.");
     }
-    
-    MixpanelError(@"Encountered an uncaught exception. All Mixpanel instances were archived.");
-    
     if (handler.defaultExceptionHandler) {
         // Ensure the existing handler gets called once we're finished
         handler.defaultExceptionHandler(exception);
     }
 }
 
+static void mp_handleSignal(int sig, siginfo_t *info, void *context) {
+    unregisterFatalSignals();
+    MixpanelError(@"We received a signal: %d", sig);
+    NSLog(@"??????");
+
+    NSException* exception = [NSException
+                              exceptionWithName:@"UncaughtException"
+                              reason:
+                              [NSString stringWithFormat:
+                               NSLocalizedString(@"Signal %d was raised.", nil),
```

Comment:
Please break after the first colon and align other colons. Xcode can do this with `Control + I`